context.go

```diff
@@ -483,3 +513,7 @@ func (ctx *cbContext) DeferCommit() func(err error) {
 		})
 	}
 }
+
+func (ctx *cbContext) addDeferred(def func() error) {
```

Comment:
Why pass the func and not the `io.Closer`?

Reply:
same reason as above, to have generic implementation. Might be changed because you're basically right :)